### ISSUE TITLE
Create a README file to easily set up the project.

### ISSUE THREAD

**Issue by shahzad713** [openpredictionmarkets/socialpredict#45]

**Comment by shahzad713** [comment]
@pwdel @jg-you

**Comment by pwdel** [comment]
Greetings @shahzad713 ... thank you so much for your comment.  So currently, I'm working on some major overhauls which would allow a user to actually sell shares and implement some more fair market mechanics.  Currently what you would get on the platform now if you deployed would look something like this:

https://brierfoxforecast.ngrok.app/markets

Can you tell me a bit more about your use case and where you are looking to deploy?  I would be happy to tailor a README.md specifically for you at this point so you can get something up and going. However, given that I'm so early on in the development, it would be helpful to understand and know that you are definitely interested in this and definitely going to deploy it some where so I can prioritize that.

Thanks again so much for your interest, happy to hear more feedback or get more issues from you.

**Comment by shahzad713** [comment]
Thanks for your reply
As I have see yours post on Upwork where you are looking for an react js
developer to collaborate with you on this project. And I am happy to
collaborate with you on this
Actually I was not able to setup this project on my local to debug it that
way I wanted about the read me file

**Comment by pwdel** [comment]
OK well thank you so much for the initiative and for reaching out on here.

I would say at this point, my priority needs to be in fixing the back end code first, unless there is someone who would actually use this software in a deployment. I work a full time job so this is on the side, so I have a limited time to contribute, only nights and weekends and here and there during lunch and what not essentially.

I would say, if you could flag me your profile on UpWork so I can find it that would be great. I actually got 141 proposals so I'm trying to start filtering through right now and that might take a while.

As far as deploying on local, the rough way to do it would be:

1. Ensure you have Docker installed on your local machine first. I'm assuming linux or MacOs here, windows is not supported.
2. Download the repo.
3. Go into the backend/ directory on your terminal and run ./builddockerfile and let the file build and tag https://github.com/openpredictionmarkets/socialpredict/blob/main/backend/builddockerfile
4. Go into the frontend/ directory on your terminal and run ./builddockerfile and let the file build and tag https://github.com/openpredictionmarkets/socialpredict/blob/main/frontend/builddockerfile
5. Go into the nginx/ directory on your terminal and run ./builddockerfile and let it build and tag. https://github.com/openpredictionmarkets/socialpredict/blob/main/nginx/builddockerfile
6. Uncomment out these lines in order to be able to seed the database. https://github.com/openpredictionmarkets/socialpredict/blob/c52ad85ee20cc2ab347598db543fb29ad05c45d9/backend/main.go#L32 ... e.g. backend/main.go the area where it lists, "seed database"
7. Navigate back to the root directory socialpredict/ and then run in your terminal ./compose-dev ... this will spin up the service on Docker and open a port. You can then visit the service at localhost:8089 and log in with either of the following users:
```
username: user1, password: password
username: user2, password: password
```
8. You can ping the API at localhost:8086 ... API end points can be viewed at here: https://github.com/openpredictionmarkets/socialpredict/blob/main/backend/server/server.go#L33  and follow the pattern: `localhost:8086/api/v0/markets` for example.
9. Note that any time you make a change to the backend, unfortunately since it's not built very well, it will make new seeds to the database if you don't comment out these lines, mentioned above. https://github.com/openpredictionmarkets/socialpredict/blob/c52ad85ee20cc2ab347598db543fb29ad05c45d9/backend/main.go#L32  ... so you have to comment them out, otherwise the graphs will start to look odd, because new bets keep getting entered.
10. If you have any problems, revert back to the earlier release, v0.0.2, though that doesn't have all of the latest features on main.
11. To spin the service down, you can go to your terminal and navigate to socialpredict and then run ./compose-dev-down which will destroy the databases as well, e.g. a full service shut down. If you do, `docker-compose down` in that directory it will maintain your database.

If you decide to go through this above, please know that this is not a guarantee to get the job...I don't want you to end up doing free work. This is just me trying to be a good steward and answer your question. However if you do end up following the above and it works, please let me know and I can formalize it into documentation for the whole internet to be able to use.

Thanks much again for your kindness and opening up this issue.

**Comment by shahzad713** [comment]
Awesome, thanks for the info!
I will try to setup the project it it work then we can create the brief
document on this
Thanks again for responding 😊

**Comment by avalonprod** [comment]
You can setting Docker network for secure connection and for running unit test you can use CI/CD.
I think you need to configure a limit on requests and more scalable authorization with tokens.

**Comment by shahzad713** [comment]
Thanks for the tip!

**Comment by pwdel** [comment]
> You can setting Docker network for secure connection and for running unit test you can use CI/CD. I think you need to configure a limit on requests and more scalable authorization with tokens.

First off thank you for your comment. Please open up a new issue for this if you are interested in socialpredict, if you would be so kind, thank you.

**Comment by shahzad713** [comment]
@pwdel @jg-you  
I am able to run all the services on my local machine now. Should I create a document on it so that others can understand things easily?

**Comment by shahzad713** [comment]
> First off thank you for your comment. Please open up a new issue for this if you are interested in socialpredict, if you would be so kind, thank you.

Yes these stop woking but we need more things to setup the porject

**Comment by pwdel** [comment]
@shahzad713 

> @pwde  I am able to run all the services on my local machine now. Should I create a document on it so that others can understand things easily?

Thank you so much for the offer. So first off, I want to make it clear that contributing to this project does not guarantee getting the job. I did not expect that people would jump from UpWork on to this repo and start contributing. I want to be respectful of your time.

On the other hand, if you do go forward with any help on this, obviously you could go forward and put that in your open source contribution portfolio and I would endorse that.

Again, I just don't want to lead you on, I have a long way to go in selecting the job and getting things going, re: the UpWork job. The needs of the project might not fit your profile.

**Comment by shahzad713** [comment]
Don't worry I wanted to contribute my knowledge on this project. It's not
necessary to have job to work here don't worry about it I spend my time
here to gain more more knowledge and experience
Thanks
